kernel/thread_abort.c

```diff
@@ -37,13 +37,17 @@ void _impl_k_thread_abort(k_tid_t thread)
 	_k_thread_single_abort(thread);
 	_thread_monitor_exit(thread);
 
-	if (_current == thread) {
-		_Swap(key);
-		CODE_UNREACHABLE;
-	}
+	if (_is_in_isr()) {
+		irq_unlock(key);
```

Comment:
we are just unlocking here and returning, where is reschedule is being called?

the earlier logic is now moved to ela=se part (non isr), so how this call will behave in case of ISR?

Reply:
The reschedule doesn't get called.  That's the whole point: in an ISR, you can't _Swap() on xtensa (and ARM too) because you then won't get to the interrupt exit code, which will then inspect the scheduler and return into whatever the next thread should be.  This is all explained in the commit message detailing that change.

Reply:
got it :), the reschedule will happen anyways after servicing the interrupt.